oauth2/handler.go

```diff
@@ -549,8 +549,16 @@ func (h *Handler) TokenHandler(w http.ResponseWriter, r *http.Request) {
 	var ctx = r.Context()
 
 	accessRequest, err := h.r.OAuth2Provider().NewAccessRequest(ctx, r, session)
+
 	if err != nil {
-		x.LogError(r, err, h.r.Logger())
+		switch err.Error() {
+		case fosite.ErrServerError.Error():
+		case fosite.ErrTemporarilyUnavailable.Error():
```

Comment:
Golang does not automatically fallthrough, you need to add the fallthrough statement to do so. In general I think you could simplify this as:

```
if errors.Is(err, fosite.ErrUnauthorizedError) {
  x.LogAudit(r, err, h.r.Logger())
} else {
  x.LogError(r, err, h.r.Logger())
}
```

Reply:
sorry about the fallback, guess that shows how much go experience i have ...

the problem with using `ErrRequestUnauthorized` (I assume this is what you mean, there doesn't seem to be an `ErrUnauthorizedError`?) is that that's not actually what's returned, we get something more specific like `ErrInvalidClient`, `ErrUnsupportedGrantType` or similar.

Reply:
No problem, it is a good learning experience :)

The error I meant was `fosite. ErrUnauthorizedClient` and `fosite.ErrInactiveToken`

Reply:
the error comparisons won't return true though. one could potentially check e.g.  `strings.Compare(err.Error(), fosite.ErrInvalidClient.Name) == 0` instead but not sure that that's the nicest solution.

regardless, i still think that anything that doesn't correspond to a status code >= 500 should be passed to the audit log rather than the errror log.

Reply:
You would want to use `errors.Cause(err) == fosite.Err...` from https://github.com/pkg/errors#retrieving-the-cause-of-an-error